clients/spark/core/src/main/scala/io/treeverse/clients/GCBackupAndRestore.scala

```diff
@@ -113,6 +114,39 @@ object GCBackupAndRestore {
     textFilePath
   }
 
+  def isPathExist(path: String, configMapper: ConfigMapper): Boolean = {
+    val conf = configMapper.configuration
+    val fs = FileSystem.get(URI.create(path), conf)
+    var pathExists = false
+    try {
+      fs.getFileStatus(new Path(path))
+      pathExists = true
+    } catch {
+      case e: FileNotFoundException => print(path + " not exists.")
+      case e: IOException           => e.printStackTrace()
+    }
+    pathExists
+  }
+
+  /** Eliminate objects that don't exist on the underlying object store from the path list.
+   *  @param absolutePathsDF a data frame containing object absolute paths
+   *  @param hc hadoop configurations
+   *  @return a dataset only including absolute paths of objects that exist on the underlying object store
+   */
+  def eliminatePathsOfNonExistingObjects(
+      absolutePathsDF: Dataset[String],
+      hc: Configuration
+  ): Dataset[String] = {
+    // Spark operators will need to generate configured FileSystems to check if objects exist.
+    // They will not have a JobContext to let them do that. Transmit (all) Hadoop filesystem configuration values to
+    // let them generate a (close-enough) Hadoop configuration to build the
+    // needed FileSystems.
+    val hcValues =
+      spark.sparkContext.broadcast(HadoopUtils.getHadoopConfigurationValues(hc, "fs."))
+    val configMapper = new ConfigMapper(hcValues)
+    absolutePathsDF.filter(x => isPathExist(x, configMapper))
```

Comment:
Why not use `fs.exists`?

```suggestion
    absolutePathsDF.filter(x => {
      val p = new Path(x)
      p.getFileSystem(configMapper.configuration).exists(p)
    }
```

Reply:
I also think that to unify our usage we should get the FS from the path, and also `exists` saves a few lines of code, but you should catch `IOException`, and it might be still implemented under the `isPathExist` method.